bin/gem_downloader

```diff
@@ -20,7 +20,27 @@ Dir.chdir(gp.target_dir) {
   gp.target_files.each { |n, v, an|
     path = sprintf(file_format, index, n, v)
     loop {
-      `curl -o #{path} -L http://rubygems.org/downloads/#{n}-#{v}.gem`
+      # Fetch the pre-compiled gem native extensions instead of building them
+      # from source if the version has prefix `gem-fetch-`.
+      if v.start_with?('gem-fetch-')
+        v.slice!('gem-fetch-')
+        actual_version = v.dup
+        fail if actual_version.nil?
```

Comment:
Isn't this
```ruby
fail unless actual_version
```
? But, more importantly, are we really expecting `v.dup` to return `nil`?

Reply:
This line no longer exists after the refactoring.